storages/python/python-storage-plugins/mongodb.py

```diff
@@ -0,0 +1,92 @@
+import json
+import sys
+from urllib.parse import quote_plus, urlparse
+
+import pymongo
+
+import dlite
+from dlite.options import Options
+from dlite.utils import instance_from_dict
+
+
+class mongodb(dlite.DLiteStorageBase):
+    """DLite storage plugin for PostgreSQL."""
+    def open(self, uri, options=None):
+        """Opens `uri`.
+
+        The `options` argument provies additional input to the driver.
+        Which options that are supported varies between the plugins.  It
+        should be a valid URL query string of the form:
+
+            key1=value1;key2=value2...
+
+        An ampersand (&) may be used instead of the semicolon (;).
+
+        Options:
+        - database: Name of database to use (default: "dlite")
+        - collection: Name of collection to use (default: "dlite_coll")
+        - user: User name.
+        - password: Password.
+        - mode: "r" for opening in read-only mode, "w" for read/write mode.
+        - authMechanism: Authentication mechanism (default: "SCRAM-SHA-256")
+        - mock: Whether to use mongomock.
+        """
+        opts = Options(
+            options,
+            defaults='database=dlite;collection=dlite_coll;mode=w;'
+            'authMechanism=SCRAM-SHA-256;mock=no',
+        )
+        opts.setdefault('password', None)
+        self.writable = True if 'w' in opts.mode else False
+
+        # Connect to existing database
+        user = quote_plus(opts.user) if opts.user else None
+        password = quote_plus(opts.password) if opts.password else None
+
+        if dlite.asbool(opts.mock):
+            import mongomock
+            mongo_url = urlparse(f'mongodb://{uri}')
+            port = mongo_url.port if mongo_url.port else 27017
+
+            @mongomock.patch(servers=((mongo_url.hostname, port), ))
+            def get_client():
+                return open_client()
+        else:
+            def get_client():
+                return open_client()
+
+        def open_client():
+            client = pymongo.MongoClient(
+                host=uri,
```

Comment:
Do you have an example of what `uri` might look like?
In PyMongo terms, the Mongo URI is the full URI, e.g., `mongodb://localhost:27017/`, where `host` would then be `localhost` in this case.

Here, you are using the `uri` parameter as the `host` (+ `port`), which clashes with the PyMongo definition.

This is most likely fine, but it would be good with a `uri` parameter example in the method's doc string or a reference to a more expanded documentation of how this parameter may look like?
This whole thing will e.g., _not_ work if I include the scheme part of the URI when using the mongodb storage backend...

Reply:
Good point. `location` would probably be a better name for the `uri` argument. It might be worth adding an issue for changing it consistently throughout all plugins?

If the user wants to fetch an instance from a MongoDB storage, he or she could write

```python
inst = Instance.from_url('mongodb://localhost:27017?user=me#4f3f2898-69e3-5a0d-820a-c4e87987dbce')
```

which would result in the following three calls to the mongodb plugin: 

```python
mongodb.open(uri='localhost:27017', options='user=me')
mongodb.load(id='4f3f2898-69e3-5a0d-820a-c4e87987dbce')
mongodb.close()
```

Reply:
An issue should be raised for this to improve the legibility of the MongoDB plugin code.

Reply:
Not sure what happened with my previous reply. But I have now added issue #325 as well as an additional point to issue #80. Merging now...